src/Compiler/Microsoft.CodeAnalysis.Razor/src/BindTagHelperDescriptorProvider.cs

```diff
@@ -244,171 +238,163 @@ static TagHelperDescriptor CreateFallbackBindTagHelper()
-                {
-                    var (assemblyName, typeName, namespaceName) = displayNames.GetNames();
-
-                    results.Add(new ElementBindData(
-                        assemblyName,
-                        typeName,
-                        namespaceName,
-                        type.Name,
-                        (string)constructorArguments[0].Value,
-                        null,
-                        (string)constructorArguments[1].Value,
-                        (string)constructorArguments[2].Value,
-                        (string)constructorArguments[3].Value));
-                }
-                else if (constructorArguments.Length == 4 && SymbolEqualityComparer.Default.Equals(attribute.AttributeClass, bindInputElement))
-                {
-                    var (assemblyName, typeName, namespaceName) = displayNames.GetNames();
-
-                    results.Add(new ElementBindData(
-                        assemblyName,
-                        typeName,
-                        namespaceName,
-                        type.Name,
-                        "input",
-                        (string)constructorArguments[0].Value,
-                        (string)constructorArguments[1].Value,
-                        (string)constructorArguments[2].Value,
-                        (string)constructorArguments[3].Value));
-                }
-                else if (constructorArguments.Length == 6 && SymbolEqualityComparer.Default.Equals(attribute.AttributeClass, bindInputElement))
-                {
-                    var (assemblyName, typeName, namespaceName) = displayNames.GetNames();
-
-                    results.Add(new ElementBindData(
-                        assemblyName,
-                        typeName,
-                        namespaceName,
-                        type.Name,
-                        "input",
-                        (string)constructorArguments[0].Value,
-                        (string)constructorArguments[1].Value,
-                        (string)constructorArguments[2].Value,
-                        (string)constructorArguments[3].Value,
-                        (bool)constructorArguments[4].Value,
-                        (string)constructorArguments[5].Value));
+                    // We need to check the constructor argument length here, because this can show up as 0
+                    // if the language service fails to initialize. This is an invalid case, so skip it.
+                    if (constructorArguments.Length == 4 && SymbolEqualityComparer.Default.Equals(attribute.AttributeClass, bindElementAttribute))
+                    {
+                        var (typeName, namespaceName) = displayNames.GetNames();
+
+                        tagHelper = CreateElementBindTagHelper(
+                            typeName,
+                            namespaceName,
```

Comment:
Consider using named arguments for `typeName` and `typeNamespace` as well.

Reply:
I'm not super keen on a style enforcement across the repo that insists on using named arguments for everything in case someone reorders parameters by hand. Happy to add named arguments when it makes the calling code clearer, but future proofing users from making super silly mistakes when they choose not to use tooling that is available to them.

Reply:
If you feel strongly about adding named arguments everywhere, we should probably have a larger conversation outside of the Razor repo.

Reply:
Don't feel strongly at all, thanks for clarifying the reasoning.